crates/factors/src/factor.rs

```diff
@@ -11,7 +13,7 @@ pub trait Factor: Any + Sized {
     type AppState: Default;
 
     /// The [`FactorInstancePreparer`] for this factor.
-    type InstancePreparer: FactorInstancePreparer<Self>;
+    type InstancePreparer: Default;
```

Comment:
Question: (realizing that this suggestion moves a bit back in the direction we just came from):

Would it make sense for `Factor` to only have two associated types `AppState` and `InstancePreparer`? Then the `InstancePreparer` would implement the `InstancePreparer` trait which has one associated type `InstanceState` and one method `fn prepare(self) -> anyhow::Result<Self::InstanceState>;`.

Reply:
Done.